New CreateEdit documentation (#7424)
marmelab/react-admin: docs/Create.md

```jsx
// in src/posts.js
import * as React from "react";
import { Create, SimpleForm, TextInput, DateInput, required } from 'react-admin';
import RichTextInput from 'ra-input-rich-text';

export const PostCreate = () => (
    <Create>
        <SimpleForm>
            <TextInput source="title" validate={[required()]} fullWidth />
            <TextInput source="teaser" multiLine={true} label="Short description" />
            <RichTextInput source="body" />
            <DateInput label="Publication date" source="published_at" defaultValue={new Date()} />
        </SimpleForm>
    </Create>
);

// in src/App.js
import * as React from "react";
import { Admin, Resource } from 'react-admin';
import jsonServerProvider from 'ra-data-json-server';

import { PostCreate } from './posts';

const App = () => (
    <Admin dataProvider={jsonServerProvider('https://jsonplaceholder.typicode.com')}>
        <Resource name="posts" create={PostCreate} />
    </Admin>
);

export default App;
```

You can customize the `<Create>` component using the following props:

* [`actions`](#actions): override the actions toolbar with a custom component

> * [`aside`](#aside-component): component to render aside to the main content

Reviewer: ```suggestion
* [`aside`](#aside): component to render aside to the main content
```

* `children`: the components that renders the form
* `className`: passed to the root component
* [`component`](#component): override the root component

> * [`disableAuthentication`](#disable-authentication): disable the authentication check

Reviewer: ```suggestion
* [`disableAuthentication`](#disableauthentication): disable the authentication check
```

## `redirect`

By default, submitting the form in the `<Create>` view redirects to the `<Edit>` view.

You can customize the redirection by setting the `redirect` prop to one of the following values:

- `'edit'`: redirect to the Edit view (the default)
- `'list'`: redirect to the List view
- `'show'`: redirect to the Show view
- `false`: do not redirect

```jsx
const PostCreate = () => (
    <Create redirect="list">
        ...
    </Create>
);
```

Note that the `redirect` prop is ignored if you set [the `mutationOptions` prop](#mutationoptions). See that prop for how to set a different redirection path in that case. 

If you want to allow the user to enter several records one after the other, setting `redirect` to `false` won't make it, as the form isn't emptied by default. You'll have to empty the form using the `mutationOptions`, and this option disables the `redirect` prop. Check [the Save And Add Another section](#save-and-add-another) for more details.

## `resource`

> Components based on `<Create>` are often used as `<Resource create>` props, and therefore rendered when the URL matches `/[resource]/create`. The `<Create>` component generates a call to `dataProvider.create()` using the ressource name from the URL by default.

Reviewer: ```suggestion
Components based on `<Create>` are often used as `<Resource create>` props, and therefore rendered when the URL matches `/[resource]/create`. The `<Create>` component generates a call to `dataProvider.create()` using the resource name from the URL by default.
```

> You can decide to use a `<Create>` component in another path, or embedded in a page using another ressource name (e.g. in a Dialog). In that canse, you can explicitely set the `ressource` name:

Reviewer: ```suggestion
You can decide to use a `<Create>` component in another path, or embedded in a page using another resource name (e.g. in a Dialog). In that case, you can explicitly set the `resource` name:
```

```jsx
// use a div as root component
const PostCreate = () => (
    <Create component="div">
        ...
    </Create>
);

// use a custom component as root component 
const PostCreate = () => (
    <Create component={MyComponent}>
        ...
    </Create>
);
```

The default value for the `component` prop is `Card`.

## `disableAuthentication`

By default, the `<Create>` component will automatically redirect the user to the login page if the user is not authenticated. If you want to disable this behavior and allow anonymous access to a creation page, set the `disableAuthentication` prop to `true`.

```jsx
const PostCreate = () => (
    <Create disableAuthentication>
        ...
    </Create>
);
```

## `mutationOptions`

You can customize the options you pass to react-query's `useMutation` hook, e.g. to override success or error side effects, by setting the `mutationOptions` prop. Refer to the [useMutation documentation](https://react-query.tanstack.com/reference/useMutation) in the react-query website for a list of the possible options.

> Let's see an example with the success side effect. By default, when the save action succeeds, react-admin shows a notification, and redirects to another page. You can override this behavior and pass custom success side effects by providing a `mutationOptions` prop with an `onSuccess` key:

Reviewer 2: ```suggestion
Let's see an example with the success side effect. By default, when the save action succeeds, react-admin shows a notification, and redirects the new record edit page. You can override this behavior and pass custom success side effects by providing a `mutationOptions` prop with an `onSuccess` key:
```

        </Create>
    );
}
```
{% endraw %}

The `onError` function receives the error from the `dataProvider.create()` call. It is a JavaScript Error object (see [the dataProvider documentation for details](./DataProviderWriting.md#error-format)).

The default `onError` function is:

```jsx
(error) => {
    notify(typeof error === 'string' ? error : error.message || 'ra.notification.http_error', { type: 'warning' });
}
```

**Tip**: If you want to have different failure side effects based on the button clicked by the user, you can set the `mutationOptions` prop on the `<SaveButton>` component, too.

## `record`

The `record` prop allows to initialize the form with non-empty values. It is exposed for consistency with the `<Edit>` component, but if you need default values, you should use the `defautValues` prop on the Form element instead.

## `redirect`

By default, submitting the form in the `<Create>` view redirects to the `<Edit>` view.

You can customize the redirection by setting the `redirect` prop to one of the following values:

- `'edit'`: redirect to the Edit view (the default)
- `'list'`: redirect to the List view
- `'show'`: redirect to the Show view

> - `false`: do not redirect

Reviewer 2: We also accept a function: `redirect((resource, id, data) => string)`

| Rule name               | Description                                                                          |
|-------------------------|--------------------------------------------------------------------------------------|
| `& .RaCreate-main`      | Applied to the main container                                                        |
| `& .RaCreate-noActions` | Applied to the main container when `actions` prop is `false`                         |
| `& .RaCreate-card`      | Applied to the child component inside the main container (MUI's `Card` by default)   |

To override the style of all instances of `<Create>` components using the [MUI style overrides](https://mui.com/customization/globals/#css), use the `RaCreate` key.

## `title`

By default, the title for the `Create` view is "Create [resource_name]".

You can customize this title by specifying a custom `title` prop:

```jsx
export const PostCreate = () => (
    <Create title="New post">
        ...
    </Create>
);
```

> The `title` value can be a string or a React element.

Reviewer 2: Might be worth reminding users they can access the record that was passed by calling `useRecordContext`.

Author: in the context of the Create page, it doesn't make much sense IMHO

**Tip**: The `<Create>` component also watches the "source" parameter of `location.search` (the query string in the URL) in addition to `location.state` (a cross-page message hidden in the router memory). So the `CreateRelatedCommentButton` could also be written as:

{% raw %}
```jsx
import * as React from "react";
import Button from '@mui/material/Button';
import { Link } from 'react-router-dom';

const CreateRelatedCommentButton = ({ record }) => (
    <Button
        component={Link}
        to={{
            pathname: '/comments/create',
            search: `?source=${JSON.stringify({ post_id: record.id })}`,
        }}
    >
        Write a comment for that post
    </Button>
);
```
{% endraw %}

Should you use the location `state` or the location `search`? The latter modifies the URL, so it's only necessary if you want to build cross-application links (e.g. from one admin to the other). In general, using the location `state` is a safe bet.

And if you want to prefill the form with constant values, use the `defaultValues` prop on the Form tag.  

> 

Reviewer 2: Might be worth mentioning you can use the `CloneButton` and provide its `record` prop for that

import Button from '@mui/material/Button';
import { Link } from 'react-router-dom';

const CreateRelatedCommentButton = ({ record }) => (
    <Button
        component={Link}
        to={{
            pathname: '/comments/create',
            search: `?source=${JSON.stringify({ post_id: record.id })}`,
        }}
    >
        Write a comment for that post
    </Button>
);
```
{% endraw %}

Should you use the location `state` or the location `search`? The latter modifies the URL, so it's only necessary if you want to build cross-application links (e.g. from one admin to the other). In general, using the location `state` is a safe bet.

And if you want to prefill the form with constant values, use the `defaultValues` prop on the Form tag.  

## Save And Add Another

Whe users need to create several records in a row, a good UX is to stay on the Create form after a successfull submission, and to empty that form to allow a new entry. 

Setting the `<Create redirect={false}>` prop only solves part of the problem: the form still needs to be emptied. That's why the right implementation for this use case is to use the `mutationOptions` prop:

```jsx
import * as React from 'react';

> import { useNotify, useRedirect, Create, SimpleForm } from 'react-admin';

Reviewer: ```suggestion
import { useNotify, Create, SimpleForm } from 'react-admin';
```

const PostCreate = () => {
    const { reset } = useFormContext();
    const notify = useNotify();

>     const  onSuccess: () => {

Reviewer: ```suggestion
    const  onSuccess = () => {
```
